src/NUnitEngine/nunit.engine.core/Runners/DirectTestRunner.cs

```diff
@@ -148,7 +148,7 @@ protected override TestEngineResult LoadPackage()
 #else
                 IFrameworkDriver driver = driverService.GetDriver(testFile, skipNonTestAssemblies);
 #endif
-                driver.ID = TestPackage.ID;
+                driver.ID = subPackage.ID;
```

Comment:
Right! The two will end up being the same IFF there is only one assembly in the package, which is why the error slipped through. IIRC I had to change it in the GUI engine as well.